applicationset/controllers/clusterprofileeventhandler.go

```diff
@@ -0,0 +1,170 @@
+package controllers
```

Comment:
the logic in this file seems complex and I wonder if its a copy/paste from the existing. in that case, any chance the existing logic could be edited (made polymorphic?) to allow for ClusterProfile and avoid the duplication?
I'm afraid otherwise maintaining of the two would be hard.
If the plan is to move fully to ClusterProfile (ambitious but possible), I think this file kept as is is fine and we should review it properly.